source/opt/decoration_manager.cpp

```diff
@@ -144,41 +113,45 @@ void DecorationManager::AnalyzeDecorations() {
 
   // For each group and instruction, collect all their decoration instructions.
   for (ir::Instruction& inst : module_->annotations()) {
-    switch (inst.opcode()) {
-      case SpvOpDecorate:
-      case SpvOpDecorateId:
-      case SpvOpMemberDecorate: {
-        auto const target_id = inst.GetSingleWordInOperand(0u);
+    AddDecoration(&inst);
+  }
+}
+void DecorationManager::AddDecoration(ir::Instruction* inst) {
+  switch (inst->opcode()) {
+    case SpvOpDecorate:
+    case SpvOpDecorateId:
+    case SpvOpMemberDecorate:
+    case SpvOpName: {
```

Comment:
SpvOpName looks wrong here.  That would be a functional change.

Reply:
Perhaps I'm confused, but isn't SpvOpName another decoration?  There's also SpvOpMemberName.  Yes, this is new wrt existing code, so if these are indeed decorations, perhaps they could be added later.

Reply:
The OpName are not in the annotations section.  They are part of the debug information.  SpvOpMemberName add a name for a type.  It does not take an general result id.  I was debating whether or not I should get the DecorationManager to handle names.  I added this, then started to rethink this.  I forgot to remove this.  I'll do that.